### PR TITLE
dxScheduler: fix view switching when views are set as objects

### PR DESCRIPTION
**Pull request by Yarovaya**

### CODE REVIEW

File: testing/tests/DevExpress.ui.widgets.scheduler/timeline.tests.js [hunk 1]
```diff
@@ -1169,6 +1169,22 @@ QUnit.test("TimelineWorkWeek has right count of cells with view option intervalC
...
+        currentDate: new Date(2017, 5, 25),
+        startDayHour: 8,
+        endDayHour: 20
+    });
+
+    var $element = this.instance.element();
+
+    var coords = this.instance.getCoordinatesByDate(new Date(2017, 6, 6, 12, 0), 0, false);
+
+    assert.equal(coords.top, $element.find(".dx-scheduler-date-table tbody td").eq(200).position().top, "Cell coordinates are right");
```
Comment (SergeyAlexeev) [inline testing/tests/DevExpress.ui.widgets.scheduler/timeline.tests.js]: Extract a variable for this `$element.find(".dx-scheduler-date-table tbody td").eq(200).position()`

File: testing/tests/DevExpress.ui.widgets.scheduler/workSpace.tests.js [hunk 2]
```diff
@@ -3129,6 +3129,22 @@ QUnit.testStart(function() {
...
+            currentDate: new Date(2017, 5, 25),
+            startDayHour: 8,
+            endDayHour: 20
+        });
+
+        var $element = this.instance.element();
+
+        var coords = this.instance.getCoordinatesByDate(new Date(2017, 6, 6, 12, 0), 0, false);
+
+        assert.equal(coords.top, $element.find(".dx-scheduler-date-table tbody td").eq(88).position().top, "Cell coordinates are right");
```
Comment (SergeyAlexeev) [inline testing/tests/DevExpress.ui.widgets.scheduler/workSpace.tests.js]: The same here - `$element.find(".dx-scheduler-date-table tbody td").eq(88).position()`

File: testing/tests/DevExpress.ui.widgets.scheduler/common.tests.js [hunk 3]
```diff
@@ -1104,6 +1104,29 @@ QUnit.testStart(function() {
...
+                name: "DAY1",
+                intervalCount: 5
+            }, {
+                type: "week",
+                name: "WEEK1",
+                intervalCount: 3
+            }]
+        });
+
+        var workSpaceWeek = this.instance.element().find(".dx-scheduler-work-space").dxSchedulerWorkSpaceWeek("instance"),
```
Comment (SergeyAlexeev) [inline testing/tests/DevExpress.ui.widgets.scheduler/common.tests.js]: Use this.instance.getWorkSpace instead.

File: js/ui/scheduler/ui.scheduler.timeline.js [hunk 4]
```diff
@@ -288,7 +288,7 @@ var SchedulerTimeline = SchedulerWorkSpace.inherit({
             fullDays = Math.floor(fullInterval / (toMs("day"))),
             tailDuration = fullInterval - (fullDays * toMs("day")),
             tailDelta = 0,
-            cellCount = this._getCellCountInDay() * fullDays,
+            cellCount = this._getCellCountInDay() * (fullDays - 2 * this._getWeekendsCount(fullDays)),
```
Comment (SergeyAlexeev) [inline js/ui/scheduler/ui.scheduler.timeline.js]: The `_getWeekendsCount` returns `0`, so should we call it here?

Comment (Yarovaya) [reply js/ui/scheduler/ui.scheduler.timeline.js]: _getIntervalBetween is the base method which call _getWeekendsCount.
_getWeekendsCount method is  redefined for the timelineWorkWeek

**Review by SergeyAlexeev — APPROVED** [review]

**Pull request merged**